src/data.js

```diff
@@ -0,0 +1,17 @@
+const data = (() => {
+  const getWeather = async (city, unit) => {
+    const KEY = process.env.API_KEY;
+    const weather = await fetch(
+      ` https://api.openweathermap.org/data/2.5/weather?q=${city}&units=${unit}&APPID=${KEY}`,
+      { mode: 'cors' },
+    );
+    const weatherData = await weather.json();
+    return weatherData;
+  };
+
+  return {
+    getWeather,
+  };
+})();
```

Comment:
Nice done using IIFE and the revealing module pattern.

OPTIONAL: Since you are using Webpack to modularize and bundle your code, you could simplify this and let Webpack take care of the modules/scope:

```js
const getWeather = async (city, unit) => { ... };

export default { getWeather };
```

Take a look at `dist/main.js`, to see what webpack is doing for you: Modules get wrapped into functions isolating their scope.

> Each module is a piece of code that is executed once it is loaded. In that code, there may be declarations (variable declarations, function declarations, etc.). **By default, these declarations stay local to the module.**
You can mark some of them as exports, then other modules can import them. 

https://exploringjs.com/es6/ch_modules.html#_browsers-scripts-versus-modules
https://medium.com/webpack/brief-introduction-to-scope-hoisting-in-webpack-8435084c171f

Reply:
This is interesting. I didn't know it. So I will not have to use IIFEs since I use webpack, right?

Reply:
Yes, as long as you are using ES6 modules (in this case through Webpack): https://webpack.js.org/concepts/why-webpack/.

Reply:
Thank you. I will read more and practice with it.